src/DurableSDK/PowerShellServices.cs

```diff
@@ -6,50 +6,167 @@
 namespace Microsoft.Azure.Functions.PowerShellWorker.Durable
 {
     using System;
+    using System.Collections.ObjectModel;
+    using System.Linq;
     using System.Management.Automation;
-    using PowerShell;
+    using System.Reflection.Metadata;
+    using Microsoft.Azure.Functions.PowerShellWorker.PowerShell;
+    using Microsoft.Azure.Functions.PowerShellWorker.Utility;
+    using Microsoft.Azure.WebJobs.Script.Grpc.Messages;
+    using Newtonsoft.Json;
+    using LogLevel = WebJobs.Script.Grpc.Messages.RpcLog.Types.Level;
 
     internal class PowerShellServices : IPowerShellServices
     {
-        private const string SetFunctionInvocationContextCommand =
-            "Microsoft.Azure.Functions.PowerShellWorker\\Set-FunctionInvocationContext";
-
         private readonly PowerShell _pwsh;
-        private bool _hasSetOrchestrationContext = false;
+        private bool _hasInitializedDurableFunctions = false;
+        private bool _usesExternalDurableSDK = false;
+        private readonly ErrorRecordFormatter _errorRecordFormatter = new ErrorRecordFormatter();
+        private readonly ILogger _logger;
+
+        private const string _setFunctionInvocationContextCommandTemplate = "{0}\\Set-FunctionInvocationContext";
+
+        // uses built-in SDK by default
+        private string SetFunctionInvocationContextCommand = string.Format(
+            _setFunctionInvocationContextCommandTemplate,
+            Utils.InternalDurableSdkName);
 
-        public PowerShellServices(PowerShell pwsh)
+        public PowerShellServices(PowerShell pwsh, ILogger logger)
         {
             _pwsh = pwsh;
+            _logger = logger;
+
+            // Configure FunctionInvocationContext command, based on the select DF SDK
+            var prefix = Utils.InternalDurableSdkName;
+            SetFunctionInvocationContextCommand = string.Format(_setFunctionInvocationContextCommandTemplate, prefix);
+        }
+
+        public bool isExternalDurableSdkLoaded()
+        {
+            // Search for the external DF SDK in the current session
+            var matchingModules = _pwsh.AddCommand(Utils.GetModuleCmdletInfo)
+                .AddParameter("FullyQualifiedName", Utils.ExternalDurableSdkName)
+                .InvokeAndClearCommands<PSModuleInfo>();
+
+            // If we get at least one result, we know the external SDK was imported
+            var numCandidates = matchingModules.Count();
+            var isModuleInCurrentSession = numCandidates > 0;
+
+            if (isModuleInCurrentSession)
+            {
+                var candidatesInfo = matchingModules.Select(module => string.Format(
+                    PowerShellWorkerStrings.FoundExternalDurableSdkInSession, module.Name, module.Version, module.Path));
+                var externalSDKModuleInfo = string.Join('\n', candidatesInfo);
+
+                if (numCandidates > 1)
```

Comment:
Could you please add a test case that validates this logic?

Reply:
Discussed offline, sharing update:

The methods in `PowerShellServices` aren't easy to test and given that this logic is behind a feature flag, we feel it should be safe to release as is. I have created [this](https://github.com/Azure/azure-functions-powershell-worker/issues/899) ticket to track the work of refactoring `PowerShellServices` so that unit testing these methods is easier. Prior to GA'ing this work, we will need to have unit tests for these methods.

FYI @michaelpeng36